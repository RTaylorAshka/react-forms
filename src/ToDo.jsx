import { useState } from 'react';
import Task from './Task';
import NewTaskForm from './NewTaskForm';

function ToDo() {

    const [tasks, setTasks] = useState([]);

    function removeTask(idx) {
        setTasks(tasks.filter((t, i) => i != idx));
    }

    function addTask(desc) {
        setTasks([...tasks, desc])
    }

    return (
        <div>
            <h1>To Do's</h1>
            <NewTaskForm addTask={addTask} />
            <ul>
                {tasks.map((t, idx) => <Task desc={t} key={idx} removeMe={() => removeTask(idx)} />)}

            </ul>
        </div>
    )
}




export default ToDo;