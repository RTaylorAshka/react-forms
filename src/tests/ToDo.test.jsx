import { render, fireEvent, screen } from "@testing-library/react";
import ToDo from "../ToDo";

it("should render ToDo without crashing", () => {
    render(<ToDo />);
})

it("matches snapshot", () => {
    const { asFragment } = render(<ToDo />)
    expect(asFragment()).toMatchSnapshot();
})


//NEEDS BUGFIXING
it("should add a new task", () => {
    const { queryByText, getByLabelText } = render(<ToDo />)
    expect(queryByText("Walk the fish")).not.toBeInTheDocument();
    const input = getByLabelText('New Task:', { exact: false });
    console.log('INPUTTTTTT: ', input);
    const button = queryByText("add");
    fireEvent.change(input, { target: { value: 'Walk the fish' } });
    fireEvent.click(button);
    expect(queryByText("Walk the fish")).toBeInTheDocument();




})