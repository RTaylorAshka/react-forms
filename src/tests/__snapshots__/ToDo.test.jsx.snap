// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      To Do's
    </h1>
    <form>
      <label
        for="task"
      >
        New Task:
      </label>
      <input
        id="task"
        name="task"
        placeholder="walk the cat"
        value=""
      />
      <button>
        add
      </button>
    </form>
    <ul />
  </div>
</DocumentFragment>
`;
